articles/secret-key-cryptography-in-js/index.md

```diff
@@ -0,0 +1,138 @@
+
+- **Web crypto API**. It provides the functionalities of the crypto library to web based applications. The functionalities includes encryption, decryption and generation of hashing signatures. It employs the same algorithms employed by the crypto module.
+
+- **Stanford JavaScript Crypto Library(sjcl)**. It is a project developed by computer security lab of Standard with an aim of coming up with a cryptographic library that is faster and smaller. It is considered small but powerful. It employs  the AES algorithm.
+
+### Implementing secret key cryptography in JavaScript.
+
+For purposes of this article, we shall use the crypto module which is a core module in `Node.js` so you won't have to install any third-party module for the project.
+
+#### Forming the Puzzle.
+
+For us to implement secret key cryptography in JavaScript, we need to form a practical puzzle. Let's assume, we have two communicating parties, David and Viktoria. For secret key cryptography, if David wants to send a message to Viktoria, he encrypts the message using the secret key to get the cipher text. The cipher text is transported over a secure medium whereas the secret key is transported over an insecure medium. On reaching the other end, Viktoria uses her secret key to decode the message and get the plain text. She shall follow the same process if she intends to send a reply.
+
+#### Defining variables and modules
+
+```javascript
+    const crypto = require('crypto');
+
+    const alg = `aes-192-cbc`;
+
+    const pass_key = `david_viktoria`;
+
+    const salt = 'salt';
+
+    const keylen = 24;
+
+    const shared_key = crypto.scryptSync(pass_key,salt,keylen);
+
+    const one_time_code = crypto.randomBytes(16);
+
+    const cipher = crypto.createCipheriv(alg,shared_key,one_time_code);
+
+    const decipher = crypto.createDecipheriv(alg,shared_key,one_time_code);
+```
+From the above, we define the module crypto which is a core module meaning. From there, we specify the algorithm we are going to use since crypto can support more than one algorithm. We shall use `AES`. We then come up with the neccessary parameters to get the shared key. A one time code is also needed in generating the cipher and the decipher. Based on the above, we have everything for encryption and decryption.
+
+#### David encrypting the message
+
+```javascript
+    function davidEncrypting(){
+    const davids_message = "Hello there";
+
+    cipher.on('readable',() => {
+        let _cipher_text = cipher.read()
+        if(_cipher_text){
+
+            cipher_text = _cipher_text.toString('hex');
+
+        }
+    });
+
+    cipher.write(davids_message);
+
+    cipher.end();
+   
+    };
+```
```

Comment:
make sure indentation is consistent, like your project code file: https://github.com/mwangiKibui/secret-key-cryptography-in-js/blob/main/secret-key-cryptography.js